### ISSUE TITLE
Test process is killed by failed test

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1210]
The below code causes the Test to stop prematurely because the test process died.
Also, I don't think expectedly failed tests should really display error message boxes or break on failure when `Break on Failure` isn't checked, maybe this is another bug.

```cs
using System;

namespace BeefBugRepros
{
	static
	{
		public static Result<void, int32> RetErr(ref int a)
		{
			a = 10;
			return .Err(10);
		}

		[Test(ShouldFail=true)]
		public static void TestBug()
		{
			int r = 0;
			RetErr(ref r);
		}

		[Test]
		public static void TestBug2()
		{
			// never reached
			NOP!();
		}
	}
}
```

Output:
```
Testing 'BeefBugRepros.TestBug'
ERROR: Unhandled error in result:
 10
Setting SEH filter 00007FF67CA3C240
Creating thread 73396
ERROR: Failed communicate with test target 'D:\BeefLang\BeefBugRepros/build/Test_Win64//.exe'
CrashCatcher::Crash
BEEF FATAL ERROR
Unhandled error in result:
 10Exiting thread 187268
Exiting thread 186412
Exiting thread 73396
Exiting thread 196340
Process terminated. ExitCode: 1.
Detached debugger
```

**Comment by bfiete** [comment]
Fixed at 6c0b329d57353175ce0c357a93db81f01e5c2f2d

**Comment by disarray2077** [comment]
@bfiete Now it's not breaking on failure at all, did I perhaps misunderstand something or is this another bug?

**Comment by bfiete** [comment]
Break on Failure was really supposed to mean "Break on Unexpected Failure". Meaning- if you have a bunch of tests that are unexpectedly failing but you want to set a breakpoint on not-the-first-one then you can disable Break on Failure.

**Comment by bfiete** [comment]
Technically that still won't allow you to _crash_ in a prior test, it just lets things like `Test.Assert` and `Test.FatalError` not cause a program break

**Comment by disarray2077** [comment]
> Break on Failure was really supposed to mean "Break on Unexpected Failure". Meaning- if you have a bunch of tests that are unexpectedly failing but you want to set a breakpoint on not-the-first-one then you can disable Break on Failure.

I supposed that much, it really wouldn't make sense to break on expected failures, what I meant is that it isn't breaking even in unexpected ones, you can see this by yourself if you remove the `ShouldFail=true` in the code of this issue.

**Comment by bfiete** [comment]
Ah, yes, you're right. Fixed at 344b8d01818be06b336186041c85993f4368f2f6
